testfixtures/shouldwarn.py

```diff
@@ -36,9 +36,12 @@ class ShouldWarn(warnings.catch_warnings):
 
     _empty_okay = False
 
-    def __init__(self, *expected: WarningOrType, **filters):
+    def __init__(self, *expected: WarningOrType, ordered=True, **filters):
         super(ShouldWarn, self).__init__(record=True)
-        self.expected = [C(e) for e in expected]
+        if ordered:
+            self.expected = [C(e) for e in expected]
+        else:
+            self.expected = S(*[C(e) for e in expected], ordered=ordered)
```

Comment:
Why do we need the `if ordered:`? This should work:
`self.expected = SequenceComparison(*[C(e) for e in expected], ordered=ordered)`
(feels like the full name would be clearer here, feel free to do the same with `C` -> `Comparison`!)

Reply:
This indeed does work, but it also changes error messages in existing test cases. I wanted to go with the minimal change possible, not to break anyone's tests, in case they depend on the error messages being produced.

I guess that may be a bit overcautious, because `testfixtures` error messages probably cannot be considered a stable API. I'll change the code in a way you propose.

Reply:
I used `SequenceComparison` everywhere. As noted before, this required changing some of the tests to reflect new error messages and also some of the docs examples (I have to say I really like this `Sybil` feature, I'll probably use it somewhere myself :slightly_smiling_face:).

I also realized that we need to check for `self.expected.expected` instead of `self.expected` now because empty `SequenceComparison` evaluates to `True` and does not provide `__len__`. I think it would be nice if `bool(SequenceComparison()) is False` and `len(SequenceComparison()) == 0` were available, but that's definitely out of scope of this PR.

Reply:
Yeah, `Sybil`'s there to keep the examples accurate :-)

I just refactored things a little, and renamed the parameter to match an existing one on `LogCapture`, so the second paragraph stuff isn't a problem.